### DISCUSSION TITLE
Is there something like `casadi.sum`?

### DISCUSSION THREAD

**Discussion by mrrezaie** [casadi/casadi#3526] [Q&A] [answered]
Hi,
(Happy new year!!!)
First of all, thanks for this amazing package. I'm new to CasADi; roughly speaking, I want to minimized a weighted cubed variable and I need `casadi.sum`:

```python
opt= casadi.Opti()
x = opt.variable(200,100)
...
# weight is a numpy.ndarray object with similar shape
obj = x**3 * weight 
opt.minimize( casadi.sum2(casadi.sum1(obj)) )
```
I found `casadi.sumsqr` but it doesn't fit. As you can see in the example, the combination of  `casadi.sum1` and `casadi.sum2` is working. I wonder if there is a better alternative.

Thank you in advance,
-Mohammadreza

**Comment by jgillis** [comment, marked as answer]
Indeed this would be very useful. Planned in #2320
